[2024-04-05T21:17:28.147+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: dag_postgre_operator.alter_table_task scheduled__2024-03-23T00:00:00+00:00 [queued]>
[2024-04-05T21:17:28.155+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: dag_postgre_operator.alter_table_task scheduled__2024-03-23T00:00:00+00:00 [queued]>
[2024-04-05T21:17:28.156+0000] {taskinstance.py:2193} INFO - Starting attempt 2 of 2
[2024-04-05T21:17:28.170+0000] {taskinstance.py:2217} INFO - Executing <Task(PostgresOperator): alter_table_task> on 2024-03-23 00:00:00+00:00
[2024-04-05T21:17:28.176+0000] {standard_task_runner.py:60} INFO - Started process 4925 to run task
[2024-04-05T21:17:28.179+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'dag_postgre_operator', 'alter_table_task', 'scheduled__2024-03-23T00:00:00+00:00', '--job-id', '1145', '--raw', '--subdir', 'DAGS_FOLDER/dag_python_operator_extract.py', '--cfg-path', '/tmp/tmpfrskwkky']
[2024-04-05T21:17:28.182+0000] {standard_task_runner.py:88} INFO - Job 1145: Subtask alter_table_task
[2024-04-05T21:17:28.227+0000] {task_command.py:423} INFO - Running <TaskInstance: dag_postgre_operator.alter_table_task scheduled__2024-03-23T00:00:00+00:00 [running]> on host 7f6ff1f67725
[2024-04-05T21:17:28.297+0000] {taskinstance.py:2513} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='dag_postgre_operator' AIRFLOW_CTX_TASK_ID='alter_table_task' AIRFLOW_CTX_EXECUTION_DATE='2024-03-23T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-03-23T00:00:00+00:00'
[2024-04-05T21:17:28.299+0000] {sql.py:276} INFO - Executing: 
    ALTER TABLE shooting
    ADD CONSTRAINT incident_pk PRIMARY KEY (incident_ID);
    
[2024-04-05T21:17:28.307+0000] {base.py:83} INFO - Using connection ID 'postgres_webik' for task execution.
[2024-04-05T21:17:28.315+0000] {base.py:83} INFO - Using connection ID 'postgres_webik' for task execution.
[2024-04-05T21:17:28.371+0000] {sql.py:457} INFO - Running statement: 
    ALTER TABLE shooting
    ADD CONSTRAINT incident_pk PRIMARY KEY (incident_ID);
    , parameters: None
[2024-04-05T21:17:28.398+0000] {taskinstance.py:2731} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 444, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 414, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/operators/sql.py", line 282, in execute
    output = hook.run(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 404, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 462, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.InvalidTableDefinition: multiple primary keys for table "shooting" are not allowed

[2024-04-05T21:17:28.405+0000] {taskinstance.py:1149} INFO - Marking task as FAILED. dag_id=dag_postgre_operator, task_id=alter_table_task, execution_date=20240323T000000, start_date=20240405T211728, end_date=20240405T211728
[2024-04-05T21:17:28.416+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 1145 for task alter_table_task (multiple primary keys for table "shooting" are not allowed
; 4925)
[2024-04-05T21:17:28.434+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-04-05T21:17:28.516+0000] {taskinstance.py:3312} INFO - 0 downstream tasks scheduled from follow-on schedule check
